packages/backend/src/tokens/request.ts

```diff
@@ -23,6 +23,31 @@ import {
 } from './interstitialRule';
 import type { VerifyTokenOptions } from './verify';
 
+export interface RequestAdapter {
+  /**
+   * The headers method returns the value of the specified header. Keys given to this method are
+   * normalized to lowercase, as they exist in `packages/backend/src/constants.ts`.
+   * Implementing this method requires the user to manipulate the header keys accordingly.
```

Comment:
Having the caller re-implement the headers spec is risky. Small discrepancies can cause serious issues that are very hard to debug when deployed. 

Also, we've been bitten by runtimes that don't respect the spec (Vercel runtime for edge API routes, it was fixed about 8 months ago by Vercel).

Reply:
@nikosdouvlis We did not change the implementation for our supported frameworks. We just moved the logic inside the RequestAdapter.

Reply:
Sorry, I'm still a bit confused. I'm referring to this comment:
> Implementing this method requires the user to manipulate the header keys accordingly

We can use clerk/backend to convert a `Record<string,string>` to a Headers instance without doing that manually. Is this not the preferred approach? Otherwise the method caller needs to reimplement Headers themselves